src/common/dpar_stub.c

```diff
@@ -0,0 +1,262 @@
+#include <stdio.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <pthread.h>
+#include <dlfcn.h>
+#include <inttypes.h>
+#include <string.h>
+#include <daos/dpar.h>
+
+struct par_stubs {
+	int	(*ps_init)(int *argc, char ***argv);
+	int	(*ps_fini)(void);
+	int	(*ps_barrier)(void);
+	int	(*ps_rank)(int *rank);
+	int	(*ps_size)(int *size);
+	int	(*ps_reduce)(const void *sendbuf, void *recvbuf, int count, enum par_type type,
+			      enum par_op op, int root);
+	int	(*ps_gather)(const void *sendbuf, void *recvbuf, int count, enum par_type type,
+			     int root);
+	int	(*ps_allreduce)(const void *sendbuf, void *recvbuf, int count, enum par_type type,
+				enum par_op op);
+	int	(*ps_allgather)(const void *sendbuf, void *recvbuf, int count, enum par_type type);
+	int	(*ps_bcast)(void *buffer, int count, enum par_type type, int root);
+};
+
+static struct par_stubs	 stubs;
+static void		*stubs_handle;
+
+#define FOREACH_PAR_SYMBOL(ACTION, arg)	\
+	ACTION(init, arg)		\
+	ACTION(fini, arg)		\
+	ACTION(barrier, arg)		\
+	ACTION(rank, arg)		\
+	ACTION(size, arg)		\
+	ACTION(reduce, arg)		\
+	ACTION(gather, arg)		\
+	ACTION(allreduce, arg)		\
+	ACTION(allgather, arg)		\
+	ACTION(bcast, arg)
+
+#define LOAD_SYM(name, fail)							\
```

Comment:
(style)  do {} while (0) macros should not be semicolon terminated

Reply:
They should when they are part of a foreach